packages/devtools/lib/src/timeline/flame_chart_canvas.dart

```diff
@@ -0,0 +1,624 @@
+
+    void createChartNodes(CpuStackFrame stackFrame, int row) {
+      final double width =
+          totalWidth * stackFrame.cpuConsumptionRatio - stackFramePadding;
+      final left = calculateLeftForStackFrame(stackFrame);
+      final top = (row * rowHeightWithPadding + flameChartTop).toDouble();
+
+      final node = FlameChartNode(
+        Rect.fromLTRB(left, top, left + width, top + rowHeight),
+        nextColor(),
+        Colors.black,
+        Colors.black,
+        stackFrame,
+      );
+
+      rows[row].nodes.add(node);
+
+      for (CpuStackFrame child in stackFrame.children) {
+        createChartNodes(
+          child,
+          row + 1,
+        );
+      }
+    }
+
+    createChartNodes(data.cpuProfileRoot, 0);
+  }
+
+  void selectNodeAtOffset(Offset offset) {
+    final node = getNode(offset);
+
+    // Do nothing if the tap did not occur on any nodes or if the tap was to
+    // select the already selected node.
+    if (node == null || node == selectedNode) {
+      return;
+    }
+
+    // Un-select the currently selected node if there is one.
+    if (selectedNode != null) {
+      selectedNode.selected = false;
+    }
+    // Select the new selected node.
+    node.selected = true;
+    selectedNode = node;
+
+    _stackFrameSelectedController.add(node.stackFrame);
+  }
+
+  num getFlameChartWidth() {
+    return rows[0].nodes[0].rect.right - _flameChartInset;
+  }
+
+  FlameChartNode getNode(Offset offset) {
+    final int rowIndex = getRowIndexForY(offset.dy);
+    if (rowIndex < 0 || rowIndex >= rows.length) {
+      return null;
+    }
+    return getNodeInRow(rowIndex, offset.dx);
+  }
+
+  FlameChartNode getNodeInRow(int rowIndex, double x) {
+    final row = rows[rowIndex];
+    final nodes = row.nodes;
+
+    FlameChartNode binarySearchForNode() {
+      int min = 0;
+      int max = nodes.length;
+      while (min < max) {
+        int mid = min + ((max - min) >> 1);
+        final node = nodes[mid];
+        if (x >= node.rect.left && x <= node.rect.right) {
+          return node;
+        }
+        if (x < node.rect.left) {
+          max = mid;
+        }
+        if (x > node.rect.right) {
+          min = mid + 1;
+        }
+      }
+      return null;
+    }
+
+    return nodes.isEmpty ? null : binarySearchForNode();
+  }
+
+  int getRowIndexForY(double y) {
+    if (y < flameChartTop) {
+      return -1;
+    }
+    return math.max((y - flameChartTop) ~/ rowHeightWithPadding, 0);
+  }
+}
+
+class FlameChartCanvas extends FlameChart {
+  FlameChartCanvas({
+    @required CpuProfileData data,
+    @required flameChartWidth,
+    @required flameChartHeight,
+  }) : super(
+          data: data,
+          flameChartWidth: flameChartWidth,
+          flameChartHeight: flameChartHeight,
+        ) {
+    _viewportCanvas = ViewportCanvas(
+      paintCallback: _paintCallback,
+      onTap: _onTap,
+      classes: 'ui-details-section cpu-flame-chart',
+    )..element.element.style.overflow = 'hidden';
```

Comment:
doubt the overflow hidden is needed. if it is, there is a different bug. I expect this was added when the size of the canvas was wrong so you needed to add it to make the sizes look right.

Reply:
if it is needed, add this to the main ViewPort canvas class.